### PR TITLE
Added child() method to Key class, could be used to create keys for s…

### PR DESCRIPTION
**Pull request by olegmoz**
…ubdirectories, files in directories etc

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#33](https://github.com/artipie/asto/issues/33) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/9513e67e-c445-4672-bce3-94d5bb6e3b6a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/asto/Key.java [hunk 1]
```diff
@@ -42,6 +43,13 @@
      */
     String string();
 
+    /**
+     * Creates child key with given name.
+     * @param name Child name.
+     * @return Child key.
+     */
+    Key child(String name);
```
Comment (Sammers21) [inline src/main/java/com/artipie/asto/Key.java]: `Key.From(base, name)` could be used to do the same thing.

Comment (olegmoz) [reply src/main/java/com/artipie/asto/Key.java]: Sorry, missed this constructor. Still, I believe that introducing this new child() method makes it more intuitive to create child keys from root node. I might be wrong, so it's up to you to accept it or not.

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/asto/issues/33#issuecomment-581200403))

@carlosmiranda This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#33](https://github.com/artipie/asto/issues/33) is [in scope](http://www.zerocracy.com/policy.html#14) for 3days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [3days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @olegmoz[/z](https://www.0crat.com/u/olegmoz): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fasto%2333%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@g4s8 can you remove this PR from scope? It's currently taking up space in my queue.

**Comment by g4s8** [comment]
@0crat refuse

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/asto/issues/33#issuecomment-582268975))

@g4s8 The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#33](https://github.com/artipie/asto/issues/33), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/asto/issues/33#issuecomment-582268991))

@g4s8 Job `gh:artipie/asto#33` is not assigned, can't get start time

**Comment by 0crat** [comment]
This pull request [#33](https://github.com/artipie/asto/issues/33) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/fbe83463-413e-4ddb-9806-5d89ec23c07a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by fanifieiev** [comment]
@g4s8 The PR seems to be closed. Why is `0crat` still assigning it to REVs???

**Comment by g4s8** [comment]
@0crat resign

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/artipie/asto/issues/33#issuecomment-582765962))

@g4s8 The user @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) resigned from [#33](https://github.com/artipie/asto/issues/33), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/asto/issues/33#issuecomment-582765977))

@g4s8 Job `gh:artipie/asto#33` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/asto/issues/33#issuecomment-582765977))

@g4s8 Pull request [#33](https://github.com/artipie/asto/issues/33) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**